float32 x
float32 y
float32 z
---
float32 theta1
float32 theta2
float32 d3



